molecule/default/tasks/apply.yml

```diff
@@ -324,6 +324,45 @@
             cpu: 100m
             memory: 100Mi
 
+    - name: Update the earlier deployment in check mode
+      k8s:
+        definition:
+          apiVersion: apps/v1
+          kind: Deployment
+          metadata:
+            name: apply-deploy
+            namespace: "{{ apply_namespace }}"
+          spec:
+            replicas: 1
+            selector:
+              matchLabels:
+                app: "{{ k8s_pod_name }}"
+            template: "{{ k8s_pod_template }}"
+        wait: yes
+        apply: yes
+      check_mode: yes
```

Comment:
Might just be completely missing the point (though we've used this pattern a lot in the past), but since we're not running this playbook in check mode, this would just test that the change applies normally, right?

Reply:
`check_mode: yes` means "when in normal mode, run this task in check mode". `check_mode: no` means "when in check mode, don't run this task"

Reply:
https://docs.ansible.com/ansible/latest/user_guide/playbooks_checkmode.html#enabling-or-disabling-check-mode-for-tasks

Reply:
GAH you're right. For some reason my brain is always completely the opposite on that point.

I liked `always_run` way better, because I can never determine what happens by intuition with `check_mode`.